### PR TITLE
Allowing other dose coefficients and adding ICRP119 neutron

### PR DESCRIPTION
**Pull request by shimwell**
# Description

This PR allows us to add additional dose conversion libraries. We can add others in the future but for now I've included ICRP119 neutron data which is on page 123 of this [report](https://www.icrp.org/docs/P%20119%20JAICRP%2041(s)%20Compendium%20of%20Dose%20Coefficients%20based%20on%20ICRP%20Publication%2060.pdf)

The ICRP 119 compared to the ICRP 116 is a slightly different, here is a plot for your interest

icrp_200

tagging @rlbarker for your interest

Fixes # (issue)
#2516

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have run [clang-format](https://docs.openmc.org/en/latest/devguide/styleguide.html#automatic-formatting) (version 15) on any C++ source files (if applicable)
- [x] I have followed the [style guidelines](https://docs.openmc.org/en/latest/devguide/styleguide.html#python) for Python source files (if applicable)
- [x] I have made corresponding changes to the documentation (if applicable)

relying on existing tests in tests/unit_tests/data_dose_coefficients.py

### CODE REVIEW

File: openmc/data/effective_dose/dose.py [hunk 1]
```diff
@@ -54,13 +71,17 @@ def dose_coefficients(particle, geometry='AP'):
...
+
+    if _DOSE_TABLES[library, particle] is None:
+        _load_dose(library=library, particle=particle)
 
     # Get all data for selected particle
-    data = _DOSE_ICRP116.get(particle)
+    data = _DOSE_TABLES[library, particle]
     if data is None:
-        raise ValueError("{} has no effective dose data".format(particle))
+        raise ValueError(f"{particle} has no effective dose data")
```
Comment (shimwell) [inline openmc/data/effective_dose/dose.py:84]:
```suggestion
        raise ValueError(f"{particle} has no effective dose data in library {library}")
```

File: openmc/data/effective_dose/dose.py [hunk 2]
```diff
@@ -2,40 +2,55 @@
...
+
+    """
+    print(f'loading {particle} {library}')
+    path = Path(__file__).parent / _FILES[library, particle]
+    data = np.loadtxt(path, skiprows=3, encoding='utf-8')
+    data[:, 0] *= 1e6   # Change energies to eV
+    _DOSE_TABLES[library, particle] = data
+
+
+def dose_coefficients(particle, geometry='AP', library='icrp116'):
```
Comment (paulromano) [inline openmc/data/effective_dose/dose.py:44]: I feel like "library" is already an overloaded term. How about "data_source" instead?

**Review by paulromano — CHANGES_REQUESTED** [review]
@shimwell Sorry it took me so long to get you a review on this one. I'm trying to understand the purpose of ICRP 119. It looks like that report is really just a compilation of data from older reports (specifically ICRP 60, which is from 1990). Would it not be better to have ICRP 60 directly be the data source? Furthermore, I've never seen any papers with people using ICRP 60 for evaluating dose -- can you point me to a work where either 60 or 119 was used? Sorry if these are dumb questions as I'm not a health physicist :smile:

File: openmc/data/effective_dose/dose.py (hunk 2 above)
Comment (shimwell) [reply openmc/data/effective_dose/dose.py]: data_source is good with me, I have changed the code accordingly.

**Comment by shimwell** [comment]
ICRP119 was my go to dose coefficient to try adding as it comes from the same places as the other one we already have so I thought it would be a simple update. I see what you are saying about it actually being ICRP60 so perhaps this is not the best one to add

I'm actually more interested in adding the same dose coefficients as FLUKA uses which would make comparing the codes easier. Could I add [these ](http://www.fluka.org/fluka.php?id=examples&sub=example4&node=6)photon coefficients instead?

**Comment by shimwell** [comment]
Closing as #3020 has implemented this idea with another library

**Pull request closed without merging**